File: core/src/main/java/inetsoft/web/binding/event/GetPredefinedNamedGroupEvent.java
/*
 * This file is part of StyleBI.
 * Copyright (C) 2024  InetSoft Technology
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
package inetsoft.web.binding.event;

import java.io.Serializable;

/**
 * Class that encapsulates the parameters for applying a selection.
 *
 * @since 12.3
 */
public class GetPredefinedNamedGroupEvent implements Serializable {
   /**
    * Get the assembly name.
    * @return assembly name.
    */
   public String getName() {
      return name;
   }

   /**
    * Set the assembly name.
    * @param name assembly name.
    */
   public void setName(String name) {
      this.name = name;
   }

   /**
    * Get the cell binding column name.
    * @return the field.
    */
   public String getColumn() {
      return field;
   }

   /**
    * Set the column name.
    * @param field the field.
    */
   public void setColumn(String field) {
      this.field = field;
   }

   private String name;
   private String field;
}

File: web/projects/portal/src/app/binding/event/get-predefined-named-group-event.ts
/*
 * inetsoft-web - StyleBI is a business intelligence web application.
 * Copyright © 2024 InetSoft Technology (info@inetsoft.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affrero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import { ViewsheetEvent } from "../../common/viewsheet-client/index";
import { DataRef } from "../../common/data/data-ref";

/**
 * Event for common parameters for composer object events.
 */
export class GetPredefinedNamedGroupEvent implements ViewsheetEvent {
   /**
    * Creates a new instance of <tt>GetPredefinedNamedGroupEvent</tt>.
    *
    * @param chartName the name of the viewsheet object.
    */
   constructor(name: string, column: string) {
      this.name = name;
      this.column = column;
   }

   private name: string;
   private column: string;
}
